File: Week_00/G20190343020259/LeetCode_1_0259.go
package G20190343020259

func twoSum(nums []int, target int) []int {
	var want = make(map[int]int)
	for k, v := range nums {
		if _, ok := want[v]; ok {
			return []int{want[v], k}
		}
		exp := target - v
		want[exp] = k
	}
	return []int{0, 0}
}

File: Week_02/G20190343020259/LeetCode_1_0259.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> expectMap = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (expectMap.containsKey(nums[i])) {
                return new int[]{expectMap.get(nums[i]), i};
            }
            int exp = target - nums[i];
            expectMap.put(exp, i);
        }
        return new int[]{0, 0};
    }
}